libmcount/dynamic.c

```diff
@@ -531,9 +533,19 @@ static void patch_patchable_func_matched(struct mcount_dynamic_info *mdi, struct
 				continue;
 		}
 
-		mcount_patch_func_with_stats(mdi, sym);
+		found = true;
+		match = match_pattern_list(map, soname, sym->name);
+		if (!match)
+			continue;
+		else if (match == 1)
+			mcount_patch_func_with_stats(mdi, sym);
+		else
+			mcount_unpatch_func(mdi, sym, NULL);
```

Comment:
IIRC I saw this change in the Ciena's work.

Reply:
Right, it looks 151fb6119d90d1834b548bb203f273430a929b0c missed fixing `patch_patchable_func_matched` along with `patch_normal_func_matched`.

Reply:
As it has other problems, I think I can merge this one first.